### ISSUE TITLE
ReferenceError with variables having dash in their name

### ISSUE THREAD

**Issue by ssbarnea** [nektos/act#315]
```
level=error msg="Unable to interpolate string 'py${{ matrix.python-version }}@${{ matrix.os }}' - [ReferenceError: 'version' is not defined]"
```

`python-version` is defined at matrix level and has special meaning for github so there is no way to change that.

Based on the error, I suspect that the parser is broken  as it ends up looking for version.

**Comment by chrispblink** [comment]
I'd like to add to this, you can get around this problem by using map/dict syntax (eg `matrix["python-version"]`) but this is actually invalid syntax and will fail to parse on GitHub Actions.  Woops!

**Comment by chrispblink** [comment]
Well... some poking around the codebase lead me to the previous discussion of this issue (#104) and the already-implemented fix (#287) in master 😅
